lib/stats.rb

```diff
@@ -158,7 +159,112 @@ def season_team_tackles(season)
+          postseason: postseason_stats}
+      end
+    end
+
+    seasonal_summaries
+  end
+
+  ##== SEASON SUMMARY HELPERS ==##
+
+  def season_stats(season_type, season_id, team_id)
+    season_stats = Hash.new{}
+
+    season_stats[:win_percentage] = win_percentage(season_type, season_id, team_id)
+    season_stats[:total_goals_scored] = total_goals_scored(season_type, season_id, team_id)
+    season_stats[:total_goals_against] = total_goals_against(season_type, season_id, team_id)
+    season_stats[:average_goals_scored] = average_goals_scored(season_type, season_id, team_id)
+    season_stats[:average_goals_against] = average_goals_against(season_type, season_id, team_id)
+
+    season_stats
+  end
+
+  # returns win percentage as a float for a given team in a given season
+  def win_percentage(season_type, season_id, team_id)
+    winning_game_count = 0
+    game_count = 0
+
+    @games_data.each do |game|
+      if game[:type] == season_type && game[:season] == season_id
+        if  game[:away_team_id] == team_id || game[:home_team_id] == team_id
+          game_count += 1
+          if game[:away_team_id] == team_id && game[:away_goals].to_i > game[:home_goals].to_i ||
+              game[:home_team_id] == team_id && game[:home_goals].to_i > game[:away_goals].to_i
+
+            winning_game_count += 1
+          end
+        end
+      end
+    end
+
+    ((winning_game_count.to_f / game_count) * 100.0).round(2)
+  end
+
+  # returns integer for total goals scored by a given team in a given season
+  def total_goals_scored(season_type, season_id, team_id)
+    total_goals_scored = 0
+
+    @games_data.each do |game|
+      if game[:type] == season_type && game[:season] == season_id
+        if game[:away_team_id] == team_id
+          total_goals_scored += game[:away_goals].to_i
+        elsif game[:home_team_id] == team_id
+          total_goals_scored += game[:home_goals].to_i
+        end
+      end
+    end
+    total_goals_scored
+  end
+
+  # returns integer for total goals scored AGAINST a given team in a given season
+  def total_goals_against(season_type, season_id, team_id)
+    total_goals_against = 0
+
+    @games_data.each do |game|
+      if game[:type] == season_type && game[:season] == season_id
+        if game[:away_team_id] == team_id
+          total_goals_against += game[:home_goals].to_i
+        elsif game[:home_team_id] == team_id
+          total_goals_against += game[:away_goals].to_i
+        end
+      end
+    end
+
+    total_goals_against
+  end
+
+  # returns a float for average goals scored by a given team in a given season
+  def average_goals_scored(season_type, season_id, team_id)
+    total_scored = total_goals_scored(season_type, season_id, team_id)
+    total_games = @games_data.count { |game| game[:type] == season_type && game[:season] == season_id && (game[:away_team_id] == team_id || game[:home_team_id] == team_id)}
```

Comment:
modifying the `#count` filter to this:
```ruby
total_games = @games_data.count { |game|
      game[:type] == season_type && game[:season] == season_id && \
        [game[:away_team_id], game[:home_team_id]].include?(team_id)
    }
``` 
can improve readability and cognitive complexity a few fold.  this revision drops 2 `==` equates. from the logic, this way adds the team ids to an array and then checks for the `team_id`.
Additionally, code lines should have a max length of ~80 and the `\` escape character allows you to move the logic down to the next line without breaking flow.

Reply:
please use this refactoring and apply it to `#average_goals_against`

Reply:
Updated accordingly, thanks for the tips!